include/fastdds/dds/core/policy/QosPolicies.hpp

```diff
@@ -952,9 +952,11 @@ public:
                                                                                        \
     };
 
-
+/// @copydoc GenericDataQosPolicy
 TEMPLATE_DATA_QOS_POLICY(UserDataQosPolicy, PID_USER_DATA)
+/// @copydoc GenericDataQosPolicy
 TEMPLATE_DATA_QOS_POLICY(TopicDataQosPolicy, PID_TOPIC_DATA)
+/// @copydoc GenericDataQosPolicy
 TEMPLATE_DATA_QOS_POLICY(GroupDataQosPolicy, PID_GROUP_DATA)
```

Comment:
Taking @MiguelCompany 's suggestion, I've tested this locally for Fast-RTPS-docs succesfully. The idea is to have a variable that is never defined anywhere, and then use the doxigen configuration file to set it to 1 in the PREDEFINED directive. This way, doxygen can see a class definition and it creates de corresponding XML file. As for the variable name, I just chose something, feel free to call it what you think is best and I'll adjust to that.
```
#ifdef DOXYGEN_DOCUMENTATION
/// @copydoc GenericDataQosPolicy
class UserDataQosPolicy {};
/// @copydoc GenericDataQosPolicy
class TopicDataQosPolicy {};
/// @copydoc GenericDataQosPolicy
class GroupDataQosPolicy {};
#endif

TEMPLATE_DATA_QOS_POLICY(UserDataQosPolicy, PID_USER_DATA)
TEMPLATE_DATA_QOS_POLICY(TopicDataQosPolicy, PID_TOPIC_DATA)
TEMPLATE_DATA_QOS_POLICY(GroupDataQosPolicy, PID_GROUP_DATA)
```

Reply:
You can check https://github.com/eProsima/Fast-RTPS-docs/pull/117